src/components/views/rooms/TchapExternalRoomHeader.tsx

```diff
@@ -0,0 +1,51 @@
+import { MatrixEvent, Room, RoomStateEvent } from 'matrix-js-sdk/src/matrix';
+import { useTypedEventEmitter } from 'matrix-react-sdk/src/hooks/useEventEmitter';
+import React, { useEffect, useState } from 'react';
+import { _t } from "matrix-react-sdk/src/languageHandler";
+
+import { TchapRoomAccessRule, TchapRoomAccessRulesEventId } from '../../../@types/tchap';
+import TchapRoomUtils from '../../../util/TchapRoomUtils';
+
+export interface IProps {
+    room?: Room;
+}
+
+export function checkExternalAllowed(room: Room): boolean {
+    return TchapRoomUtils.getTchapRoomAccessRule(room) === TchapRoomAccessRule.Unrestricted;
+}
+
+//define a function to manage internal state
+//inspired by RoomTopic.tsx
+function useExternalAllowed(room: Room): boolean {
+    //define the hook functions and initialize the state
+    const [isExternalAllowed, setExternalAllowed] = useState(checkExternalAllowed(room));
+
+    //listen to events to refresh the value if needed
+    useTypedEventEmitter(room.currentState, RoomStateEvent.Events, (ev: MatrixEvent) => {
+        console.log(ev.getType());
+        if (ev.getType() === TchapRoomAccessRulesEventId) {
+            setExternalAllowed(checkExternalAllowed(room));
+        }
+    });
+    //refresh the state after rendering
+    useEffect(() => {
```

Comment:
Hmmm.
I think the react lifecycle is : state change -> rendering -> effect.
Here the effect contains a state change (setExternalAllowed). Does that make an infinite loop ? state change -> rendering -> effect -> state change -> ...

I think that the rendering already uses isExternalAllowed to render the appropriate UI, so the effect is not necessary.

Reply:
I tried removing it, the status is still updated when you change externs status !

Reply:
nice

Reply:
> Does that make an infinite loop ? state change -> rendering -> effect -> state change -> ...
the effect is trigger only if room is updated  as mentioned in the comment

Reply:
ok to remove it, it works well without it :)